pkg/apis/rollouts/v1alpha1/types.go

```diff
@@ -361,6 +361,13 @@ type RolloutTrafficRouting struct {
 	Ambassador *AmbassadorTrafficRouting `json:"ambassador,omitempty" protobuf:"bytes,5,opt,name=ambassador"`
 	// AppMesh holds specific configuration to use AppMesh to route traffic
 	AppMesh *AppMeshTrafficRouting `json:"appMesh,omitempty" protobuf:"bytes,6,opt,name=appMesh"`
+	// Traefik holds specific configuration to use Traefik to route traffic
+	Traefik *TraefikTrafficRouting `json:"traefik,omitempty" protobuf:"bytes,7,opt,name=traefik"`
+}
+
+// TraefikTrafficRouting defines the configuration required to use Traefik as traffic router
+type TraefikTrafficRouting struct {
+	Service string `json:"service" protobuf:"bytes,1,name=service"`
```

Comment:
I think we should make it more evident that this expects a `TraefikService` name and not a Kubernetes service. Proposal:
```go
TraefikServiceName string `json:"traefikService" protobuf:"bytes,1,name=service"`
```
Please add field documentation as well.